### PR TITLE
Refactor create superadmin mode

### PR DESCRIPTION
**Pull request by longshuicy**
This PR is ready to be reviewed.

### Backend changes
- Store admin and admin mode in UserDoc instead of passing in parameters; remove most of the parameters on routes
- Made sure admin && adminMode logic is persist in all the method in `authorization_deps.py`
- Clean up the code about list all endpoints (see the criteria part)
- Fix the elasticsearch user clause. Depends can only be used asynchronously on routes somehow. TODO: research on fastAPI depends
- Refactoring/Adding new endpoints for GET/POST admin state, as well as admin_mode state. Write pytest accordingly
- Fix a typo on `/datasets/{dataset_id}/role/owner}` endpoint and `/metadata/{metadata_id}/role}` endpoint

### Frontend changes
- No need to pass in additional parameters to each endpoint.
- Clean up the admin mode logic and looks
- Listening to the adminMode change and enforce request to the backend one more time to reflect the effect of the admin mode in various components

image
image

### CODE REVIEW

File: backend/app/routers/authorization.py [hunk 1]
```diff
@@ -107,61 +110,68 @@ async def get_dataset_role(
...
+        file_id: str,
+        admin_mode: bool = Depends(get_admin_mode),
+        current_user=Depends(get_current_username),
+        role: RoleType = Depends(get_role_by_file),
+        admin=Depends(get_admin),
 ):
-    # admin is a superuser and has all the privileges
-    if admin:
+    # admin is a superuser and has all the privileges, only show if the user has turned on the admin mode
+    if admin and admin_mode:
```
Comment (longshuicy) [inline backend/app/routers/authorization.py:144]: This alone won't work because there is also role check on the dependencies `get_role_by_file`

**Comment by tcnichol** [comment]
I am noticing an error on this branch.
I populated data using the populate fake data, but when I tried to upload data through the web interface, I saw this:
Screenshot 2023-12-19 at 3 33 04 PM
Screenshot 2023-12-19 at 3 33 54 PM

**Comment by longshuicy** [comment]
> I am noticing an error on this branch. I populated data using the populate fake data, but when I tried to upload data through the web interface, I saw this: Screenshot 2023-12-19 at 3 33 04 PM Screenshot 2023-12-19 at 3 33 54 PM

For some reason I can't reproduce this issue. I resolved some merge conflict that might have something to do with codegen openapi. Could you try again when you have a chance?

**Review by ddey2 — APPROVED** [review]
Tested code. The code looks good and clean. @longshuicy thanks for cleaning it up.

**Review by lmarini — APPROVED** [review]

**Pull request merged**